File: 2018/src/test/java/com/benjamin/Day2Test.java
package com.benjamin;

import org.hamcrest.Matchers;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.hamcrest.MatcherAssert.assertThat;

public class Day2Test {

    private Day2 target;
    private static final String INPUT1 =
            "abcdef\n" +
            "bababc\n" +
            "abbcde\n" +
            "abcccd\n" +
            "aabcdd\n" +
            "abcdee\n" +
            "ababab\n";

    private static final String INPUT2 =
            "abcde\n" +
            "fghij\n" +
            "klmno\n" +
            "pqrst\n" +
            "fguij\n" +
            "axcye\n" +
            "wvxyz";

    @BeforeEach
    public void setup() {
        target = new Day2();
    }

    @Test
    public void testDeelEen1() {
        int result = target.deelEenA(INPUT1);

        assertThat(result, Matchers.is(12));
    }

    @Test
    public void testDeelEen2() {
        int result = target.deelEenB(INPUT1);

        assertThat(result, Matchers.is(12));
    }

    @Test
    public void testDeelTwee1() {
        String result = target.deelTweeA(INPUT2);

        assertThat(result, Matchers.is("fgij"));
    }
}

File: 2021/src/test/kotlin/Day2Test.kt
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

internal class Day2Test {

  private val testInput = """forward 5
down 5
forward 8
up 3
down 8
forward 2"""

  @Test
  fun `should test part1`() {
    val testInput = testInput

    assertEquals(150, Day2.part1(testInput))
  }

  @Test
  fun `should test part2`() {
    val testInput = testInput

    assertEquals(900, Day2.part2(testInput))
  }
}
